### PR TITLE
build(ci): copy CI configs

### LINKED ISSUE #3: Copy CI and mergify configs from main Cryostat project

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes #3

### CODE REVIEW

**Comment by andrewazores** [comment]
The release drafter check should start working after this is merged and the `.github/release-drafter.yml` is present in `main`.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        fi
+      if: ${{ github.event_name == 'push' && github.repository_owner == 'cryostatio' }}
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: save
+    - name: Push to quay.io
+      id: push-to-quay
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat
```
Comment (ebaron) [inline .github/workflows/ci.yaml:99]: This should be `cryostat-reports`.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - id: query-pom
+      name: Get properties from POM
+      # Query POM for core and image version and save as output parameter
+      run: |
+        CORE_VERSION="$(mvn help:evaluate -Dexpression=io.cryostat.core.version -q -DforceStdout)"
+        echo "::set-output name=core-version::v$CORE_VERSION"
+        IMAGE_VERSION="$(mvn help:evaluate -Dexpression=cryostat.imageVersion -q -DforceStdout)"
```
Comment (ebaron) [inline .github/workflows/ci.yaml:26]: This POM doesn't have a `cryostat.imageVersion` property. We should add a similar property to ensure this workflow can tag/push the built image.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        distribution: 'adopt'
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - uses: actions/download-artifact@v2
+      with:
+        name: cryostat-core
+        path: /home/runner/.m2/repository/io/cryostat/cryostat-core/
+    - run: git submodule init
+    - run: git submodule update
```
Comment (ebaron) [inline .github/workflows/ci.yaml:73]: I think these can be removed.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    - run: mvn -B -U clean verify
+    - name: Print itest logs
+      if: ${{ failure() }}
+      run: ls -1dt target/cryostat-itest-*.log | head -n1 | xargs cat
```
Comment (ebaron) [inline .github/workflows/ci.yaml:77]: I don't think this step will work.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (andrewazores) [reply .github/workflows/ci.yaml]: Thanks, fixed all four of these comments.

**Review by ebaron — APPROVED** [review]
Looks good!

**Pull request merged**